datalad/support/gitrepo.py

```diff
@@ -931,18 +935,48 @@ def __eq__(self, obj):
         """
         return self.realpath == obj.realpath
 
+    def is_valid_git(self):
+        """Returns whether the underlying repository appears to be still valid
+
+        Note, that this almost identical to the classmethod is_valid_repo(). However,
+        if we are testing an existing instance, we can save Path object creations. Since this testing
+        is done a lot, this is relevant. Creation of the Path objects in is_valid_repo() takes nearly half the time of
+        the entire function.
+
+        Also note, that this method is bound to an instance but still class-dependent, meaning that a subclass
+        cannot simply overwrite it. This is particularly important for the call from within __init__(),
+        which in turn is called by the subclasses' __init__. Using an overwrite would lead to the wrong thing being
+        called.
+        """
+
+        return self._dot_git.exists() and (
+            not self._dot_git.is_dir() or self._valid_git_test_path.exists()
+        )
+
     @classmethod
     def is_valid_repo(cls, path):
         """Returns if a given path points to a git repository"""
-        path = Path(path) / '.git'
+        if not isinstance(path, Path):
+            path = Path(path)
+        dot_git_path = path / '.git'
+
         # the aim here is to have this test as cheap as possible, because
         # it is performed a lot
         # recognize two things as good-enough indicators of a present
-        # repo: 1) a non-empty .git directory (#3473) and 2) a pointer
-        # file or symlink
-        return path.exists() and (
-            not path.is_dir() or \
-            any(path.iterdir()))
+        # repo: 1) a non-empty .git directory (#3473)
```

Comment:
@mih, @kyleam : Please double-check the change to `is_valid_repo`. In PR #3475 you changed it before and wrote a test for creation (the second half of which might now fail). As I test for not just "not empty", I'm not quite clear how `create` should behave in the test situation and how the validity of the repo actually changes the outcome.

Reply:
The test was just documenting the limits of the validity check.  Now that you do more, I see no issue with updating the test to account for the new behavior.

Reply:
Okay, thanks!